unixfs/unixfs.go

```diff
@@ -258,3 +259,23 @@ func BytesForMetadata(m *Metadata) ([]byte, error) {
 func EmptyDirNode() *dag.ProtoNode {
 	return dag.NodeWithData(FolderPBData())
 }
+
+// ValidatePB validates a unixfs protonode.
+func ValidatePB(n *dag.ProtoNode, pb *pb.Data) error {
+	if len(pb.Blocksizes) == 0 { // special case
+		return nil
```

Comment:
Can probably still check if data size is less-equal than the declared filesize.

Reply:
There's also the case where there *are* no blocks but we still have inline data.

Reply:
> There's also the case where there are no blocks but we still have inline data.

And in that case I don't think there is anything to validate, but I could be mistaken.

Reply:
Actually I take that back.  I will have a closer look later today.